examples/basic-async/basic-async.c

```diff
@@ -33,15 +34,30 @@ main(void)
 	 */
 	for (int i = 0; i < 2; i++) {
 		char *src1 = malloc(TEST_SIZE * sizeof(char));
```

Comment:
if this were a test I wouldn't complain, but in an example, I think this is just too bleh.
I think we can create a function from the inner parts of this loop, and use goto error handling to manage the src/dst buffers.